Discord/src/packetHandlers/handlers/EventsHandlers.ts

```diff
@@ -158,23 +178,161 @@ export default class EventsHandlers {
+			lng,
+			moneyLost: packet.moneyLost
+		});
+
+		if (packet.guildPointsLost > 0) {
+			desc += i18n.t("models:players.leavePVEIslandMalusGuildPoints", {
+				lng,
+				guildPointsLost: packet.guildPointsLost
+			});
+		}
+
+		desc += i18n.t("models:players.leavePVEIslandDescEnd", {
+			lng
+		});
+
+		await interaction.channel.send({
+			embeds: [
+				new DraftBotEmbed()
+					.formatAuthor(i18n.t("models:players.leavePVEIslandTitle", {
+						lng,
+						pseudo: interaction.user.displayName
+					}), interaction.user)
+					.setDescription(desc)
+			]
+		});
 	}
 
 	@packetHandler(PlayerLevelUpPacket)
-	async playerLevelUp(_context: PacketContext, _packet: PlayerLevelUpPacket): Promise<void> {
-		// Todo
+	async playerLevelUp(context: PacketContext, packet: PlayerLevelUpPacket): Promise<void> {
+		const interaction = DiscordCache.getInteraction(context.discord!.interaction);
+
+		if (!interaction) {
+			return;
+		}
+
+		const lng = interaction.userLanguage;
+
+		const rewards: { [key: string]: { tr: string, replacements?: object } } = {
+			healthRestored: { tr: "healthRestored" },
+			fightUnlocked: { tr: "fightUnlocked" },
+			guildUnlocked: { tr: "guildUnlocked" },
+			classesTier1Unlocked: { tr: "classTier", replacements: { tier: 1 } },
+			classesTier2Unlocked: { tr: "classTier", replacements: { tier: 2 } },
+			classesTier3Unlocked: { tr: "classTier", replacements: { tier: 3 } },
+			classesTier4Unlocked: { tr: "classTier", replacements: { tier: 4 } },
+			classesTier5Unlocked: { tr: "classTier", replacements: { tier: 5 } },
+			missionSlotUnlocked: { tr: "newMissionSlot" },
+			pveUnlocked: { tr: "pveUnlocked" },
+			statsIncreased: { tr: "statsIncreased" }
+		};
+
+		let desc = i18n.t("models:players.levelUp.description", {
+			lng,
+			level: packet.level
+		});
+
+		for (const [key, value] of Object.entries(rewards)) {
+			if (packet[key as keyof PlayerLevelUpPacket]) {
+				desc += `${i18n.t(`models:players.levelUp.rewards.${value.tr}`, {
+					lng,
+					...value.replacements
+				})}\n`;
+			}
+		}
+
+		await interaction.channel.send({
+			embeds: [
+				new DraftBotEmbed()
+					.formatAuthor(i18n.t("models:players.levelUp.title", {
+						lng,
+						pseudo: interaction.user.displayName
+					}), interaction.user)
+					.setDescription(desc)
+			]
+		});
 	}
 
 	@packetHandler(PlayerReceivePetPacket)
-	async playerReceivePet(_context: PacketContext, _packet: PlayerReceivePetPacket): Promise<void> {
-		// Todo
+	async playerReceivePet(context: PacketContext, packet: PlayerReceivePetPacket): Promise<void> {
+		const interaction = DiscordCache.getInteraction(context.discord!.interaction);
+
+		if (!interaction) {
+			return;
+		}
+
+		const lng = interaction.userLanguage;
+		const descTr = packet.giveInGuild
+			? "models:petReceived.genericGiveGuild"
+			: packet.giveInPlayerInv
+				? "models:petReceived.genericGivePlayer"
+				: "models:petReceived.genericGiveNoSlot";
```

Comment:
```suggestion
		const descTr = `models:petReceived.genericGive${
		packet.giveInGuild ? "Guild" : packet.giveInPlayerInv ? "Player" : "NoSlot"}`;
```

(attention j'ai ptet fait des warnings eslints sur les indents / newlines trucs du genre

Reply:
ah et aussi le déplacer directement dans la création de l'embed vu que la clé n'est utilisée qu'une seule fois

Reply:
Je préfère mettre le chemin complet car l'IDE autocomplète et surligne si la traduction est manquante. On peut aussi shift click pour aller directement dans la trad

Reply:
ah, faudra vraiment me passer ce plugin, perso quand je le fais, j'ai juste rien